lib/gst/player/gstplayer-media-info.c

```diff
@@ -0,0 +1,603 @@
+gst_player_audio_info_get_bitrate (const GstPlayerAudioInfo* info)
+{
+  g_return_val_if_fail (GST_IS_PLAYER_AUDIO_INFO (info), -1);
+
+  return info->bitrate;
+}
+
+/**
+ * gst_player_audio_info_get_max_bitrate:
+ * @info: a #GstPlayerAudioInfo
+ *
+ * Returns: the audio maximum bitrate in #GstPlayerAudioInfo.
+ */
+guint
+gst_player_audio_info_get_max_bitrate (const GstPlayerAudioInfo* info)
+{
+  g_return_val_if_fail (GST_IS_PLAYER_AUDIO_INFO (info), -1);
+
+  return info->max_bitrate;
+}
+
+/* Subtitle information */
+G_DEFINE_TYPE (GstPlayerSubtitleInfo, gst_player_subtitle_info,
+    GST_TYPE_PLAYER_STREAM_INFO);
+
+static void
+gst_player_subtitle_info_init (GstPlayerSubtitleInfo * info)
+{
+  info->language = NULL;
+}
+
+static void
+gst_player_subtitle_info_finalize (GObject * object)
+{
+  GstPlayerSubtitleInfo *info = GST_PLAYER_SUBTITLE_INFO (object);
+
+  g_free (info->language);
+
+  G_OBJECT_CLASS (gst_player_subtitle_info_parent_class)->finalize (object);
+}
+
+static void
+gst_player_subtitle_info_class_init (GObjectClass * klass)
+{
+  klass->finalize = gst_player_subtitle_info_finalize;
+}
+
+GstPlayerSubtitleInfo*
+gst_player_subtitle_info_new (GstPlayerSubtitleInfo *ref)
+{
+  GstPlayerSubtitleInfo *ret = NULL;
+
+  ret = g_object_new (GST_TYPE_PLAYER_SUBTITLE_INFO, NULL);
+
+  if (ref)
+    ret->language = g_strdup (ref->language);
+
+  return ret;
+}
+
+/**
+ * gst_player_subtitle_get_language:
+ * @info: a #GstPlayerSubtitleInfo
+ *
+ * Returns: the language of the stream, or NULL if unknown.
+ */
+gchar*
+gst_player_subtitle_get_language( const GstPlayerSubtitleInfo* info)
+{
+  g_return_val_if_fail (GST_IS_PLAYER_SUBTITLE_INFO (info), NULL);
+
+  return info->language;
+}
+
+/* Global media information */
+G_DEFINE_TYPE (GstPlayerMediaInfo, gst_player_media_info, G_TYPE_OBJECT);
+
+static void
+gst_player_media_info_init (GstPlayerMediaInfo * info)
+{
+  info->uri = NULL;
+  info->stream_list = NULL;
+}
+
+static void
+gst_player_media_info_finalize (GObject * object)
+{
+  GstPlayerMediaInfo  *info = GST_PLAYER_MEDIA_INFO (object);
+
+  g_free (info->uri);
+  info->uri = NULL;
+
+  if (info->stream_list) {
+    g_list_free (info->stream_list);
+    info->stream_list = NULL;
+  }
+
+  G_OBJECT_CLASS (gst_player_media_info_parent_class)->finalize (object);
+}
+
+static void
+gst_player_media_info_class_init (GstPlayerMediaInfoClass * klass)
+{
+  GObjectClass  *oclass = (GObjectClass *) klass;
+
+  oclass->finalize = gst_player_media_info_finalize;
+}
+
+/**
+ * gst_player_media_info_get_uri:
+ * @info: a #GstPlayerMediaInfo
+ *
+ * Returns: the URI associated with #GstPlayerMediaInfo.
+ */
+gchar *
+gst_player_media_info_get_uri (const GstPlayerMediaInfo *info)
+{
+  g_return_val_if_fail (GST_IS_PLAYER_MEDIA_INFO (info), NULL);
+
+  return info->uri;
+}
+
+/**
+ * gst_player_media_info_get_stream_list:
+ * @info: a #GstPlayerMediaInfo
+ *
+ * Returns: (transfer full) (element-type GstPlayerStreamInfo): A #GList of
```

Comment:
As we now made the GstPlayerMediaInfo immutable, we can make this (transfer none) and just directly return info->stream_list instead of copying it